include/aos/test/utils.hpp

```diff
@@ -33,6 +33,28 @@ class TestUtils {
 
         return true;
     }
+
+    template <typename T1, typename T2, typename P>
+    static bool CompareArrays(const T1 array1, const T2 array2, P predicate)
```

Comment:
implementation doesnt handle arrays like "aaabb" "bbbaa"
probably UnorderedPointwise/UnorderedElementsAreArray matchers does  something similar

Reply:
Updated UT to use UnorderedPointwise